staging/src/k8s.io/api/flowcontrol/v1beta2/types.go

```diff
@@ -352,8 +352,10 @@ type FlowSchemaStatus struct {
 	// `conditions` is a list of the current states of FlowSchema.
 	// +listType=map
 	// +listMapKey=type
+	// +patchStrategy=merge,retainKeys
+	// +patchMergeKey=type
```

Comment:
because strategic merge patches are computed client-side, changing this breaks compatibility with clients built against 1.23 and earlier versions of the API types

Reply:
also, what does the `retainKeys` bit do? I don't see that used on any other conditions lists that use the merge strategy

Reply:
Yeah, I think we should drop retainKeys.

@liggitt, adding the patchStrategy means that new clients will start sending correct patches and old clients will continue to do the wrong thing, isn't that better than having all clients do the wrong thing? Also, the wrong thing is only the wrong thing if clients send partial conditions, which I don't know if it happens.

Reply:
> adding the patchStrategy means that new clients will start sending correct patches and old clients will continue to do the wrong thing, isn't that better than having all clients do the wrong thing?

I don't think so. it also means new clients will send incorrect patches to old servers... they'll compute a merge patch, send only the changed object, and the old server will discard all other conditions.

I think this has to be changed in an API version boundary

Reply:
As the author of the only known client, let me point out that due to the astonishing lack of documentation, I expected that my client (the APF config-consuming controller) sends a strategic merge patch with just one Condition in the Status but expects that the result will only be a modification of that individual Condition; this is the bug #104842 .

Due to the stunning lack of documentation, I do not see how we can be sure any other client --- if any exists at all --- is written with the correct interpretation of the existing annotations.

BTW, I find `,retainKeys` in the example I look at, Pod.

Reply:
Of course, we certainly can not assume that all client authors are as confused as I was.  So, yes, we fix this later.

That means that for now, @tkashem proceed with your PR that changes the controller to use SSA.  In this here PR you can see what I suppose is the intended breathless style of composing the ApplyConfiguration.

I will do something about a PR that adds a currently-correct test of patching

Reply:
> As the author of the only known client ...

I was referring to the strategic merge patches that the `k8s.io/apimachinery/pkg/util/strategicpatch` package (used by things like kubectl) would generate automatically from the linked API types